File: web-app/src/lib/features/features/logs.ts
import type { Feature } from '../types';

export const logsFeatures = {
  'logs-get': {
    id: 'logs-get',
    title: 'Get Logs',
    description: 'Retrieves log output for a session by ID or title.',
    rpcIds: ['logs:get'],
    componentPaths: [],
    testIds: [
      'TestGetLogs_NoSessionID_NilPoller',
      'TestGetLogs_WithUUID_NilPoller',
      'TestGetLogs_WithUUID_MatchingInstance',
      'TestGetLogs_WithUUID_NoMatchingInstance',
      'TestGetLogs_WithTitle_FindsLogByTitle',
      'TestGetLogs_EmptySessionID',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;

File: server/features/logs.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

var LogsGet = featureregistry.Feature{
	ID:          "logs-get",
	Title:       "Get Logs",
	Description: "Retrieves log output for a session by ID or title.",
	RPCIDs:      []string{"logs:get"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(LogsGet)
}
